app/models/katello/concerns/pulp_database_unit.rb

```diff
@@ -55,85 +55,47 @@ def pulp_data(pulp_id)
         content_unit_class.new(pulp_id)
       end
 
-      # Import all units of a single type and refresh their repository associations
-      def import_all(pulp_ids = nil, options = {})
-        index_repository_association = options.fetch(:index_repository_association, true) && self.manage_repository_association
-        service_class = SmartProxy.pulp_master!.content_service(content_type)
+      def unit_id_field
+        "#{self.name.demodulize.underscore}_id"
+      end
 
-        process_block = lambda do |units|
+      # Don't know if this makes any sense anymore if we do not update repository associations?
+      def import_all(pulp_ids = nil)
```

Comment:
testing this, when uploading an rpm, it doesn't seem to get associated to the repository (the rpm count doesn't increase).   I think its because filtered_index_content.rb calls import_all() and passes in the unit ids, but that no longer associates them to the repository.

Reply:
Yes..We will need to handle that for upload content for rpm and deb if we remove the repository association part from import_all..
We could pass an optional parameter repository to import_all and then call
` sync_repository_associations(repository, :pulp_ids => pulp_ids) ` after importing units as we do in import_for_repository. We could use that in filtered_index_content.rb.

Reply:
Pushed changes based on the last few comments in a separate commit https://github.com/Katello/katello/pull/8123/commits/f15c8adc87215bfa5f4ce28a96c14e6b30f0237f